tests/test-customize-changesets-controller.php

```diff
@@ -99,29 +518,1239 @@ public function test_update_item() {
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $uuid ) );
+		$request->set_body_params( array(
+			'date_gmt' => 'BAD DATE',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'bad_customize_changeset_date', $response, 400 );
+
+		$manager = new WP_Customize_Manager();
+		$this->assertNull( $manager->find_changeset_post_id( $uuid ) );
+	}
+
+	/**
+	 * Test that update_item() rejects editing a changeset with an invalid date.
+	 */
+	public function test_update_item_edit_bad_customize_changeset_date() {
+		wp_set_current_user( self::$admin_id );
+
+		$manager = new WP_Customize_Manager();
+		$manager->save_changeset_post();
+
+		$date_before = get_post( $manager->changeset_post_id() )->post_date_gmt;
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $manager->changeset_uuid() ) );
+		$request->set_body_params( array(
+			'date_gmt' => 'BAD DATE',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'bad_customize_changeset_date', $response, 400 );
+		$this->assertSame( $date_before, get_post( $manager->changeset_post_id() )->post_date_gmt );
+	}
+
+	/**
+	 * Tests that creating a published changeset with update_item() returns a 'publish' status.
+	 */
+	public function test_update_item_create_status_is_publish_after_publish() {
+		if ( post_type_supports( 'customize_changeset', 'revisions' ) ) {
+			$this->markTestSkipped( 'Changesets are not trashed when revisions are enabled.' );
+		}
+
+		wp_set_current_user( self::$admin_id );
+
+		$uuid = wp_generate_uuid4();
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $uuid ) );
+		$request->set_body_params( array(
+			'status' => 'publish',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertSame( 200, $response->get_status() );
+
+		$data = $response->get_data();
+		$this->assertSame( 'publish', $data['status'] );
+	}
+
+	/**
+	 * Test that publishing a changeset with update_item() returns a 'publish' status.
+	 */
+	public function test_update_item_edit_status_is_publish_after_publish() {
+		if ( post_type_supports( 'customize_changeset', 'revisions' ) ) {
+			$this->markTestSkipped( 'Changesets are not trashed when revisions are enabled.' );
+		}
+
+		wp_set_current_user( self::$admin_id );
+
+		$manager = new WP_Customize_Manager();
+		$manager->save_changeset_post();
+
+		$request = new WP_REST_Request( 'PUT', sprintf( '/customize/v1/changesets/%s', $manager->changeset_uuid() ) );
+		$request->set_body_params( array(
+			'status' => 'publish',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$data = $response->get_data();
+		$this->assertSame( 'publish', $data['status'] );
+	}
+
+	/**
+	 * Test that creating a published changeset with update_item() returns a new changeset UUID.
+	 */
+	public function test_update_item_create_has_next_changeset_id_after_publish() {
```

Comment:
> I don't think it is right from a REST API design perspective to return next_changeset_uuid here since it is not part of the resource that you are making a request about.

Makes sense to me. I'll remove these tests for now.

> I suppose it would be more pure if there was something like a customize/v1/uuid endpoint that would spit out a random UUID, but that also doesn't seem completely RESTful. Maybe ultimately the generation of the UUID should be the responsibility of the JS client instead?

Placing responsibility on the client seems preferable to me to a REST endpoint for just a UUID.  How do you think it would be best to approach generating the UUID within core's JS (an existing library, from scratch, something else)? 

I suppose someone in a bind about generating the UUIDs could also create an ad-hoc admin-ajax endpoint to return a UUID from PHP.